TRANSLATION_WORKFLOW.md

```diff
@@ -0,0 +1,282 @@
+
+The workflows are:
+
+- queueing translation requests
+  - this workflow takes a PR, looks for translatable files, and queues them up for either human or machine translation
+- sending content for machine translation
+  - this workflow takes queued up machine translation requests, and submits them to the vendor
+- sending content for human translation
+  - this workflow takes queued up human translation requests, and submits them to the vendor
+- checking for finished translations, downloading content, cleaning up
+  - this workflow has two jobs (one for human, one for machine) which check a given project for finished translations, downloads translated files, and PRs them back into the repository
+
+## Workflow 1: Queueing Translation Requests
+
+This workflow looks through all of the files that are being merged into the main branch, finds the files that are of the `.mdx` type and have the `translate` frontmatter. It then adds the file names (i.e. the slugs) to a PostgreSQL database table called `translations` along with the `locale` they are to be translated to and a `status` of `PENDING`.
+
+All of the changed files will be added to this queue continuously on every merge to main with a status of `PENDING`. If a particular file is changed multiple times in this period, it is only queued if it not already queued with the `PENDING` status.
+
+As part of the machine translation work, this step also determines whether a file is meant to be human or machine translated. To do this, we look at the `translate` frontmatter, which indicates to us what is meant to be human translated. Anything not included in the `translate` frontmatter, or if the frontmatter is missing entirely, indicates to us that a document should be machine translated. Once we know that, we set the `project_id` field in the `translations` database table for the translation request record.
+
+### Manually triggering workflow
+
+There is an additional github action to be able to manually queue up specific files. This can be manually triggered in the Actions tab of github under the workflow `Manually Add Slugs to Translation Queue`. When clicking `Run workflow` you will be prompted to enter the slugs that you would like to queue up, separated by comma. An example input would be `src/content/docs/codestream/how-use-codestream/discuss-code.mdx,src/content/docs/infrastructure/host-integrations/host-integrations-list/nginx/nginx-integration.mdx`. You additionally can select to queue it up for machine translation and give a locale, such as `ja-JP, ko-KR`.
+
+### Summary
+
+- Executes: on every merge to the `main` branch`
+- Can be manually triggered? `Yes`, when using manual version of workflow
+- Steps:
+  - Looks at the files that have been changed on the merge
+  - Will save the filename, locale and status to the Translation table if not already pending
+    - `project_id` set to either machine or human translation project
```

Comment:
would it be a good idea to mention what the machine and human translation project ids are here?

Reply:
We've gotta store these in One Password cuz they're kinda 🔒